api/migrations/2020-05-30-210421_init_models/up.sql

```diff
@@ -16,24 +16,26 @@ CREATE TABLE program_specs (
     name VARCHAR(50) NOT NULL CHECK (char_length(name) > 0),
     description TEXT NOT NULL,
     hardware_spec_id UUID NOT NULL REFERENCES hardware_specs(id),
-    input SMALLINT[] NOT NULL,
-    expected_output SMALLINT[] NOT NULL,
-    UNIQUE(slug, hardware_spec_id)
+    -- TODO add constraint to make sure all values in range
+    input INT[] NOT NULL CHECK (array_length(input, 1) <= 256),
+    expected_output INT[] NOT NULL CHECK (array_length(input, 1) <= 256),
```

Comment:
I changed these from 16-bit to 32-bit. 16-bit is what the actual GDLK values are, but it's easier if the DB types match what comes in from the API, then just apply the proper range changes with constraints. Also, a signed 16-bit int can't actually hold an unsigned 16-bit so shit was a lil fucked.